### PR TITLE
Update Ruby to 2.3.2

### PR DESCRIPTION
**Pull request by adarsh**
Reason for Change
=================
* Ruby 2.3.2 was released.
* https://www.ruby-lang.org/en/news/2016/11/15/ruby-2-3-2-released/

Changes
=======
* Update Ruby version in `Gemfile`.
* Add a `.ruby-version` file, [which all Ruby switchers respect][1].
* Update the `README.md`
* Update the CircleCI configuration.

Minor
=====
* Rubocop infers the `.ruby-version` file value as the configuration for it's `TargetRubyVersion:` setting.
* In this case, to prepare for Ruby 3.0 when all strings will be frozen, it is recommended to add this "magic comment" AKA pragma to the top of each file. Luckily, `rubocop -a` does this for us.

[1]: https://gist.github.com/fnichol/1912050

### CODE REVIEW

File: spec/factories/resources.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# frozen_string_literal: true
```
Comment (ptrikutam) [inline spec/factories/resources.rb]: Seeing this pretty much everywhere. What's going on?

Comment (adarsh) [inline spec/factories/resources.rb]: See the PR comments under "Minor".

Comment (ptrikutam) [inline spec/factories/resources.rb]: Crap, didn't read the PR fully.

Comment (adarsh) [inline spec/factories/resources.rb]: > Freezing Strings feature improves apps performance by freezing Strings. So, Matz – Ruby’s creator – decided to make all String literals frozen (immutable) by default in Ruby 3.0.

> In order to have a transition path to this coming big change, it was decided to have a magic comment at the beginning of files, so you can use in Ruby 2.3.

https://blog.lucascaton.com.br/2016/01/19/what-is-frozen_string_literal-in-ruby/

Comment (ptrikutam) [inline spec/factories/resources.rb]: Alright, sounds good. Thanks for explaining.

**Pull request merged**